### ISSUE TITLE
External assets (`AssetSpec`s) show up as having "never been materialized" in grouped view (Global Asset Lineage)

### ISSUE THREAD

**Issue by tobiplay** [dagster-io/dagster#24870]
### Dagster version

1.8.9 (core)

### What's the issue?

We are using `AssetSpec`s to "override" the assets automatically created by the dlt integration. For example, if we use a source `test` with a resource `res`, Dagster automatically creates an upstream dependency named `test_res`, linking to the actual asset in the global sset lineage. By creating an `AssetSpec` with the same asset key, we can add extra metadata to this upstream dependency and make the data lineage clearer. The extra metadata may include a description, tags etc.

However, in the Dagster UI, these upstream assets show up as having “never been materialized”, even though Dagster cannot materialize these assets since they are external. This results in a visual inconsistency in the UI, as seen in the following pictures.

Upstream assets being shown as "never been materialized":
<img width="338" alt="image" src="https://github.com/user-attachments/assets/0a42d32b-37f8-402c-aa83-30ce2c817ca5">

Closer look into the group shows that they're all external:
<img width="314" alt="image" src="https://github.com/user-attachments/assets/0b365818-1849-4fce-a3e3-0bfddf7f9212">

Error when user would try to materialize these external assets:
<img width="400" alt="image" src="https://github.com/user-attachments/assets/d1e1c448-36f8-4e0d-994d-642159144830">

### What did you expect to happen?

I would expect that no information about the materialization status would be displayed for these external assets. It is easy to imagine a scenario where external and actual assets are grouped together, and in such cases, the group’s status would inaccurately reflect the true state of its materializable assets. I've tried that, and indeed the group is showing, e.g., 5/5 have never been materialized, when only 2/5 are actually materializable by Dagster.

### How to reproduce?

As explained above, by overriding dlt's upstream assets with `AssetSpec`s and grouping them, the group will then show a status for unmaterialized assets.

### Deployment type

Local

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by tobiplay** [comment]
@salazarm Fair question. I found that approach mentioned in the [docs about the dlt integration](https://docs.dagster.io/integrations/embedded-elt/dlt), starting from "[a] common question is how to define metadata on the external assets upstream of the dlt assets."

To add a bit more context, we were also planning on using the `AssetSpec`s to make the data lineage and dependencies a bit clearer, especially concerning shared API endpoint across assets.

E.g., with our implementation of a custom Dagster-dlt translator, leveraging `AssetSpec`s, both `accounting__results_yearly` and `accounting__results_ad_hoc_reporting` would point to the upstream `accounting__results` asset, with metadata along the lines of "Accounting's `results/` API endpoint to fetch the company's financial results based on user-defined parameters." 

Thereby, we'd also "disable" the automatic creation of dlt upstream assets, because they're literally just the same "box" (same asset name) without any metadata and a dashed border, upstream of the actual asset that's materializable. I don't quite understand why they're created in the first place. Is it done to show that they're part of dlt?

Then, we would just inject the common upstream asset key (API endpoint asset) via the translator's `get_deps_asset_keys` method into the dependencies of both dlt assets.

**Comment by sryza** [comment]
@salazarm yeah this is standard usage of external assets, if I'm understanding correctly: https://docs.dagster.io/concepts/assets/external-assets#external-assets.  We generally still call them "SDAs", but they're not materializable like "normal" assets.
